### DISCUSSION TITLE
How can we exchange two lines easily

### DISCUSSION THREAD

**Discussion by pengfu** [xyflow/xyflow#2723] [Q&A]
### Edge line can exsist on canvas independently without any source or target node

I have a usage scenario in a flow chart editor. Suppose a node is connected to two different target nodes, and the two connections are configured differently. If the user finds that the configuration of these two lines is reversed, he wants to exchange them. How to allow users to easily switch these two lines.

This is like exchanging the water in two containers. We may think of taking the third cup as a temp variable. But can we pour water directly into the air and exchange it ? Correspondingly, can we temporarily place a line on the canvas, as shown in the following picture.
![image](https://user-images.githubusercontent.com/7261281/187642065-0c6f4707-f876-4b77-9d51-236497894438.png)
In this situation we may exchange the two lines easily without asking temp node for help, or deleting or reconcofig any line in which case the config work may be expensive.

**Comment by moklick** [comment]
This is only possible with and edge update like this:

https://user-images.githubusercontent.com/2857535/187713037-9ac83e86-e322-41db-a987-cda21b41046b.mov

https://reactflow.dev/docs/examples/edges/updatable-edge/
